Source/WebCore/Modules/permissions/PermissionObserver.h

```diff
@@ -30,20 +30,23 @@
 
 namespace WebCore {
 
+class Page;
 class ScriptExecutionContext;
 enum class PermissionState : uint8_t;
+enum class PermissionQuerySource : uint8_t;
 struct ClientOrigin;
 struct PermissionDescriptor;
 
-class PermissionObserver : public CanMakeWeakPtr<PermissionObserver> {
+class PermissionObserver : public CanMakeWeakPtr<PermissionObserver, WeakPtrFactoryInitialization::Eager> {
```

Comment:
Why is this change needed? Are we ever constructing a WeakPtr to a PermissionObserver while not on the PermissionObserver's context thread? I couldn't find such case in this patch but I might have missed it.

Reply:
I think a previous iteration of the patch was doing this, but that is no longer the case, so I'll remove this change.